packages/atlas/src/views/studio/PlaylistWorkspace/PlaylistWorkspace.tsx

```diff
@@ -0,0 +1,299 @@
+  const mdMatch = useMediaMatch('md')
+  const addAsset = useAssetStore((state) => state.actions.addAsset)
+  const [, setThumbnailHashPromise] = useState<Promise<string> | null>(null)
+  const [showSelectDialog, setShowSelectDialog] = useState(false)
+  const [shouldFallbackThumbnail, setShouldFallbackThumbnail] = useState(true)
+
+  const { control, getValues, setValue, trigger, watch } = useForm<PlaylistWorkspaceFormFields>({
+    shouldFocusError: true,
+    defaultValues: {
+      isPublic: true,
+    },
+  })
+  const thumbnail = watch('thumbnail')
+  const { video } = useBasicVideo(playlistVideos[0] ?? '', {
+    skip: !playlistVideos[0],
+    onError: (error) =>
+      SentryLogger.error('Failed to fetch video', 'VideoTile', error, { video: { id: playlistVideos[0] } }),
+  })
+  const { url: thumbnailPhotoUrl } = useAsset(video?.thumbnailPhoto)
+
+  // const hasUnsavedAssets = dirtyFields.thumbnail?.cropId || false
+  const thumbnailAsset = useRawAsset(thumbnail?.cropId || null)
+  const originalThumbnailAsset = useRawAsset(thumbnail?.originalId || null)
+  // const firstVideoAsset = useRawAsset(video?.thumbnailPhoto?.id || null)
+
+  const computeThumbnailHash = useCallback((file: Blob) => {
+    const hashPromise = computeFileHash(file)
+    setThumbnailHashPromise(hashPromise)
+  }, [])
+
+  const updateFallbackThumbnail = useCallback(() => {
+    if (video) {
+      const currentThumbnailValue = getValues('thumbnail')
+      setValue(
+        'thumbnail',
+        {
+          ...currentThumbnailValue,
+          ...{ cropId: video.thumbnailPhoto?.id ?? null, originalId: null, url: thumbnailPhotoUrl },
+        },
+        { shouldDirty: true }
+      )
+    }
+  }, [getValues, setValue, thumbnailPhotoUrl, video])
+
+  useEffect(() => {
+    if (!thumbnailAsset) {
+      return
+    }
+    if (thumbnailAsset?.blob) {
+      computeThumbnailHash(thumbnailAsset.blob)
+    }
+  }, [computeThumbnailHash, thumbnailAsset])
+
+  useEffect(() => {
+    if (
+      video &&
+      (!thumbnail?.cropId || (thumbnail.cropId !== video.thumbnailPhoto?.id && !thumbnail.originalId)) &&
+      shouldFallbackThumbnail
+    ) {
+      setShouldFallbackThumbnail(false)
+      updateFallbackThumbnail()
+    }
+  }, [
+    video,
+    thumbnail,
+    getValues,
+    setValue,
+    thumbnailPhotoUrl,
+    shouldFallbackThumbnail,
+    updateFallbackThumbnail,
+    playlistVideos,
+  ])
+
+  const handleThumbnailFileChange = (file: ImageInputFile | null) => {
+    const currentThumbnailValue = getValues('thumbnail')
+    if (!file) {
+      if (video) {
+        setValue(
+          'thumbnail',
+          {
+            ...currentThumbnailValue,
+            ...{ cropId: video.thumbnailPhoto?.id ?? null, originalId: null, url: thumbnailPhotoUrl },
+          },
+          { shouldDirty: true }
+        )
+      }
+      setValue('thumbnail', { ...currentThumbnailValue, ...{ cropId: null, originalId: null } }, { shouldDirty: true })
```

Comment:
Do we call `setValue` here twice? If there is a `video` we will call the second `setValue` anyway, right? Should we add `else` here?  
I'm not sure if this will cause bugs.